sql/src/main/java/io/crate/executor/transport/task/KillJobTask.java

```diff
@@ -0,0 +1,109 @@
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.executor.transport.task;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import io.crate.executor.JobTask;
+import io.crate.executor.RowCountResult;
+import io.crate.executor.TaskResult;
+import io.crate.executor.transport.kill.KillJobsRequest;
+import io.crate.executor.transport.kill.KillResponse;
+import io.crate.executor.transport.kill.TransportKillJobsNodeAction;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.cluster.ClusterService;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.node.DiscoveryNodes;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class KillJobTask extends JobTask {
+
+    private final SettableFuture<TaskResult> result;
+    private final List<ListenableFuture<TaskResult>> results;
+    private ClusterService clusterService;
+    private UUID jobToKill;
+    private TransportKillJobsNodeAction nodeAction;
+
+
+    public KillJobTask(ClusterService clusterService,
+                       TransportKillJobsNodeAction nodeAction,
+                       UUID jobId,
+                       UUID jobToKill) {
+        super(jobId);
+        this.clusterService = clusterService;
+        this.nodeAction = nodeAction;
+        result = SettableFuture.create();
+        results = ImmutableList.of((ListenableFuture<TaskResult>) result);
+        this.jobToKill = jobToKill;
+    }
+
+    @Override
+    public void start() {
+        DiscoveryNodes nodes = clusterService.state().nodes();
+        KillJobsRequest request = new KillJobsRequest(ImmutableList.of(jobToKill));
+        final AtomicInteger counter = new AtomicInteger(nodes.size());
+        final AtomicLong numKilled = new AtomicLong(0);
+        final AtomicReference<Throwable> lastThrowable = new AtomicReference<>();
+
+        for (DiscoveryNode node : nodes) {
+            nodeAction.execute(node.id(), request, new ActionListener<KillResponse>() {
```

Comment:
I've just changed this method to `executeOnAllNodes` on master.. so I guess you'll have to rebase and then some code here isn't necessary anymore. Sorry about that =)